src/Avalonia.Controls.ColorPicker/Converters/ColorToHexConverter.cs

```diff
@@ -62,21 +59,159 @@ public class ColorToHexConverter : IValueConverter
+
+        /// <summary>
+        /// Converts the given color to its hex color value string representation.
+        /// </summary>
+        /// <param name="color">The color to represent as a hex value string.</param>
+        /// <param name="alphaPosition">The output position of the alpha component.</param>
+        /// <param name="includeSymbol">Whether the hex symbol '#' will be added.</param>
+        /// <returns>The input color converted to its hex value string.</returns>
+        public static string ToHexString(
+            Color color,
+            AlphaComponentPosition alphaPosition,
+            bool includeSymbol = false)
+        {
+            uint intColor;
+            if (alphaPosition == AlphaComponentPosition.Trailing)
+            {
+                intColor = ((uint)color.R << 24) | ((uint)color.G << 16) | ((uint)color.B << 8) | (uint)color.A;
+            }
+            else
+            {
+                // Default is Leading alpha
+                intColor = ((uint)color.A << 24) | ((uint)color.R << 16) | ((uint)color.G << 8) | (uint)color.B;
+            }
 
-            if (Color.TryParse(hexValue, out Color color))
+            string hexColor = intColor.ToString("x8", CultureInfo.InvariantCulture).ToUpperInvariant();
+
+            if (includeSymbol)
+            {
+                hexColor = '#' + hexColor;
+            }
+
+            return hexColor;
+        }
+
+        /// <summary>
+        /// Parses a hex color value string into a new <see cref="Color"/>.
+        /// </summary>
+        /// <param name="hexColor">The hex color string to parse.</param>
+        /// <param name="alphaPosition">The input position of the alpha component.</param>
+        /// <returns>The parsed <see cref="Color"/>; otherwise, null.</returns>
+        public static Color? ParseHexString(
+            string hexColor,
+            AlphaComponentPosition alphaPosition)
+        {
+            hexColor = hexColor.Trim();
+
+            if (!hexColor.StartsWith("#", StringComparison.Ordinal))
+            {
+                hexColor = "#" + hexColor;
+            }
+
+            if (TryParseHexFormat(hexColor.AsSpan(), alphaPosition, out Color color))
             {
                 return color;
             }
-            else if (hexValue.StartsWith("#", StringComparison.Ordinal) == false &&
-                     Color.TryParse("#" + hexValue, out Color color2))
+
+            return null;
+        }
+
+        /// <summary>
+        /// Parses the given span of characters representing a hex color value into a new <see cref="Color"/>.
+        /// </summary>
+        /// <remarks>
+        /// This is based on the Color.TryParseHexFormat() method.
+        /// It is copied because it needs to be extended to handle alpha position.
+        /// However, the alpha position enum is only available in the controls namespace with the ColorPicker control.
+        /// </remarks>
+        private static bool TryParseHexFormat(
```

Comment:
Not ideal to copy this method from Color. However, see remarks above for reasons why.